apisix/init.lua

```diff
@@ -550,6 +550,12 @@ function _M.handle_upstream(api_ctx, route, enable_websocket)
 
     set_upstream_headers(api_ctx, server)
 
+    -- lua proxy the request to upstream
+    if api_ctx.lua_proxy_upstream then
```

Comment:
I thought about the naming of the existing variables `bypass_nginx_upstream` and `lua_proxy_upstream`, and felt that their meanings are unclear. I think we can design them as two: 
* `bypass_nginx_upstream` to indicate bypassing nginx's upstream module
* `bypass_balancer` to indicate bypassing the existing balancer logic
cc @membphis

Reply:
I agree with this